contracts/examples/NFTExample.sol

```diff
@@ -75,6 +60,25 @@ contract NFTExample is
         return ConfidentialERC721URIStorage._mint(to, tokenId);
     }
 
+    function _setTokenURI(
+        uint256 tokenId,
+        ctUint64[] calldata _itTokenURI,
+        bytes[] calldata _itSignature
+    ) internal {
+        gtUint64[] memory _tokenURI = new gtUint64[](_itTokenURI.length);
+
+        itUint64 memory it;
+
+        for (uint256 i = 0; i < _itTokenURI.length; ++i) {
+            it.ciphertext = _itTokenURI[i];
+            it.signature = _itSignature[i];
+
+            _tokenURI[i] = MpcCore.validateCiphertext(it);
+        }
+
+        ConfidentialERC721URIStorage._setTokenURI(msg.sender, tokenId, _tokenURI);
```

Comment:
shouldnt you check first that the tokenId exists for this msg.sender?

Reply:
Since we are now setting the token URI during the mint tx (and only during the mint tx), we don't need to verify that the owner of this token ID is in fact the msg.sender. However, we could add this check to the ConfidentialERC721URIStorage contract if we wanted to, depending on the desired behaviour. However, I think it is better leave it to the developer to decide who can set the token URI and when. Perhaps it would make sense to add to the ConfidentialERC721URIStorage some check to ensure the token ID has been minted already.